core/backend/src/IModelHost.ts

```diff
@@ -487,10 +488,10 @@ export class IModelHost {
 
   /** This method must be called when an iModel.js services is shut down. Raises [[onBeforeShutdown]] */
   public static async shutdown(): Promise<void> {
-    if (!this._isValid)
+    if (!IModelHost._isValid)

```

Comment:
since we don't allow subclassing (the constructor is private), `IModelHost === this`. It's good to be consistent, but I see about 50/50 in this file. Personally, I'd go with `this`.

Reply:
The problem with using `this` was that we pass `IModelHost.shutdown` as a listener for the process `"beforeExit"` event, so the method becomes unbound, and `this` is no longer `IModelHost` (the _"should auto-shutdown on process beforeExit event"_ test I added confirms this - it fails without this change).

We could use `.bind()` or a lambda as the listener instead, but then we'd need to save a reference to it to be able to call `process.removeListener` at the end of `IModelHost.shutdown`, so I just thought this was easier.

Reply:
i see. Since we call `process.once` i don't think there's any reason to call `removeListner`. But, a comment about "this method is set as a listener where `this` isn't set would help. 

BTW, `BeEvent.addListener` both allows a `scope` argument and returns a removal function to avoid these problems.